src/main/redux/sagas/auth.ts

```diff
@@ -452,14 +453,38 @@ async function opdsSetAuthCredentials(
     return [, new Error("")];
 }
 
+let _implicitNonce = "";
+const setAndGetImplicitNonceForImplicitAuthentication = () => (_implicitNonce = nanoid(16), _implicitNonce);
+const getImplicitNonceForImplicitAuthentication = () => _implicitNonce;
+
 function getHtmlAuthenticationUrl(auth: IOPDSAuthDocParsed) {
 
     let browserUrl: string;
     switch (auth.authenticationType) {
 
         case "http://opds-spec.org/auth/oauth/implicit": {
 
-            browserUrl = auth.links?.authenticate?.url;
+            try {
+                const browserUrlParsed = new URL(auth.links?.authenticate?.url);
+
+                // The unique identifier used in the callback URI must have the same value as the identifier provided in the Authentication Document.
+                // https://drafts.opds.io/authentication-for-opds-1.0#342-a-shared-client-identifier:~:text=The%20unique%20identifier%20used%20in%20the%20callback%20URI%20must%20have%20the%20same%20value%20as%20the%20identifier%20provided%20in%20the%20Authentication%20Document.
+
+                // see also the callback response : it must use the id query parameter to indicate the Authentication Document identifier
+                // https://drafts.opds.io/authentication-for-opds-1.0#342-a-shared-client-identifier:~:text=it%20must%20use%20the%20id%20query%20parameter%20to%20indicate%20the%20Authentication%20Document%20identifier
+
+                browserUrlParsed.searchParams.set("client_id", encodeURIComponent_RFC3986(auth?.id || "http://opds-spec.org/auth/client"));
```

Comment:
I think using the `auth?.id` here is incorrect. The `id` specified in the OPDS Authentication Document is the unique identifier of the document itself, not the id for the client to use when authenticating.

Reply:
From the documentation:

> Example of an OPDS callback containing an Access Token
> 
> `opds://authorize/?id=http%3A%2F%2Fexample.org%2Fauth.json&access_token=9b3dc428-df5f-4bd2-9f0d-72497cbf8464&token_type=bearer`

Using the [example](https://drafts.opds.io/authentication-for-opds-1.0#233-example) as a guide, the OAuth Server should returns an `opds://authorize` redirect after successful authentication should include a querystring parameter with the opds document id in it.